pkg/sql/schemachanger/scdecomp/decomp.go

```diff
@@ -189,6 +190,9 @@ func (w *walkCtx) walkType(typ catalog.TypeDescriptor) {
 				LogicalRepresentation:  typ.GetMemberLogicalRepresentation(ord),
 			})
 		}
+	case descpb.TypeDescriptor_COMPOSITE:
+		name := tree.Name(typ.GetName())
+		panic(scerrors.NotImplementedErrorf(&name, "composite types not supported in new schema changer"))
```

Comment:
Is the schema changer support PR ready for review? I'm just being mindful of accruing technical debt so to be honest I'd rather this change were part of this PR, but won't insist on it. Hopefully removing this `scerrors.NotImplementedErrorf` error should be quite easy at this point.

Reply:
Yes, it's ready for review. I moved it out of the PR at @chengxiong-ruan suggestion to keep things small. At this point I do not want to re-add it - can we get this one merged and then I'll work on the other one immediately?

Reply:
That's fine by me, thanks! This need to change a zillion things everytime one touches anything in schema-land is sadly a fairly typical feature of our development experience.